src/kOS/AddOns/RemoteTech/Addon.cs

```diff
@@ -30,31 +31,32 @@ private void InitializeSuffixes()
         private static ScalarValue RTGetDelay(VesselTarget tgtVessel)
         {
             double waitTotal = 0;
-
-            if (RemoteTechHook.IsAvailable(tgtVessel.Vessel.id) && tgtVessel.Vessel.GetVesselCrew().Count == 0)
+            if (RemoteTechHook.IsAvailable())
             {
                 waitTotal = RemoteTechHook.Instance.GetShortestSignalDelay(tgtVessel.Vessel.id);
             }
-
-            return waitTotal;
+            return Double.IsPositiveInfinity(waitTotal) ? -1 : waitTotal;
         }
 
         private static ScalarValue RTGetKSCDelay(VesselTarget tgtVessel)
         {
             double waitTotal = 0;
 
-            if (RemoteTechHook.IsAvailable(tgtVessel.Vessel.id) && tgtVessel.Vessel.GetVesselCrew().Count == 0)
+            if (RemoteTechHook.IsAvailable() && tgtVessel.Vessel.GetVesselCrew().Count == 0)
             {
                 waitTotal = RemoteTechHook.Instance.GetSignalDelayToKSC(tgtVessel.Vessel.id);
             }
 
-            return waitTotal;
+            return Double.IsPositiveInfinity(waitTotal) ? -1 : waitTotal;
         }
 
         private static BooleanValue RTAntennaHasConnection(PartValue part)
         {
             bool result = false;
 
+            // IsAvailable(Id) is only able to return True on loaded vessels, but this
+            // is a test for a specific PART on a vessel, and individual parts on
+            // vessels don't exist when the vessel is unloaded anyway.
```

Comment:
The above comment is my justification why here the IsAvailable(id) is still being used.